### PR TITLE
Implement support for tokens from FAFs Ory Hydra

### LINKED ISSUE #434: Support OAuth tokens from Ory Hydra along with legacy tokens
The Spring Security OAuth library is deprecated and the replacement does not offer the features of an OAuth Authorisation server.
Therefore reimplemented OAuth 2.0 OpenID Connect login for FAF in the faf-user-service. However since we can't have a big-bang migration we need to have a transition path.

Therefore we need to support the current tokens as well as the new customized Ory Hydra OpenID Connect tokens.

### PR DESCRIPTION
**Pull request by Brutus5000**
Closes #434

### CODE REVIEW

File: src/main/java/com/faforever/api/security/FafMultiTokenStore.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  @Override
+  public OAuth2Authentication readAuthentication(OAuth2AccessToken token) {
+    return readAuthentication(token.getValue());
+  }
+
+  @Override
+  public OAuth2Authentication readAuthentication(String token) {
+    try {
+      return classicTokenStore.readAuthentication(token);
+    } catch (Exception e) {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/security/FafMultiTokenStore.java]: I guess it is too much effort to use more specific exceptions?

File: src/main/resources/config/application-dev.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ faf-api:
   jwt:
     secretKeyPath: ${JWT_PRIVATE_KEY_PATH:test-pki-private.key}
     publicKeyPath: ${JWT_PUBLIC_KEY_PATH:test-pki-public.key}
+    hydraJwksUrl: ${JWT_HYDRA_JWKS_URL:https://hydra.test.faforever.com/.well-known/jwks.json}
```
Comment (bukajsytlos) [inline src/main/resources/config/application-dev.yml:6]: is hydra somehow relevant in this context? isn't it implementation detail?

Comment (Brutus5000) [reply src/main/resources/config/application-dev.yml:6]: It's not necessarily Hydra that is relevant here, but it is the OpenID Connect token that we customized (manual extension for our role system). So it's not just a default Jwks url, it must point to a faf hydra instance.

File: src/main/java/com/faforever/api/security/FafMultiTokenStore.java (hunk 1 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/security/FafMultiTokenStore.java]: I'm not exactly sure which exceptions are thrown, I know about AccessDeniedException though

Comment (Brutus5000) [reply src/main/java/com/faforever/api/security/FafMultiTokenStore.java]: Browsed through the code, there is a shitload of potential Exceptions waiting to be thrown. I'll just leave it as is. It does no harm.

File: src/main/resources/config/application-dev.yml (hunk 2 above)
Comment (Brutus5000) [reply src/main/resources/config/application-dev.yml:6]: Slightly renamed it to emphasize it's faf hydra not any hydra

Comment (bukajsytlos) [reply src/main/resources/config/application-dev.yml:6]: I still don't understand, if we change hydra for octa or something else, wouldn't we need just to change the value and not the key?

**Pull request merged**